### ISSUE TITLE
Remote Containers: Execute in WSL leads to permission error.

### ISSUE THREAD

**Issue by MicahZoltu** [microsoft/vscode-remote-release#8787]
- VSCode Version: 1.80.2
- Local OS Version: Windows 11: 22H2
- Remote OS Version: Bullseye
- Remote Extension/Connection Type: Containers

- Logs:
  ```
  [488 ms] Setting up container for folder or workspace: /mnt/wsl/docker-desktop-bind-mounts/Ubuntu/d5fa517070ed97cad90f6b3aad2ba5614d2e374cfc6fd74186d8432c2e1edae3
  [494 ms] Start: Check Docker is running
  [494 ms] Start: Run in Host: docker version --format {{.Server.APIVersion}}
  [555 ms] Server API version: 1.43
  [556 ms] Start: Run in Host: docker volume ls -q
  [630 ms] Start: Run in Host: tar --no-same-owner -x -f -
  [639 ms] Error: EACCES: permission denied, lstat 'c:\Users\redacted\source\temp\node_modules\.bin\tsc'
  [639 ms] Exit code EACCES
  ```

Steps to Reproduce:

1. Install Remote Containers.
2. Enable the `Execute in WSL` setting.
3. Create a folder in Windows.
4. Open the folder in VS Code.
5. command palette > Dev Containers: Reopen in Container
6. Choose Node.js & JavaScript > Bullseye > no additional features
7. Once it is open do `npm init` and press enter through all prompts.
8. `npm install typescript`
9. File > Close Remote Connection
10. Open the local folder again.
11. command palette > Dev Containers: Reopen in Container
12. **Notice that you get an EACCES error like above**

Workaround 1: If you delete the `node_modules/.bin` folder you can now load the project in the remote container.
Workaround 2: If you disable `Execute in WSL` setting you can load the project in the remote container.

Does this issue occur when you try this locally?: No
Does this issue occur when you try this locally and all extensions are disabled?: No

----

I haven't done troubleshooting further than these steps, but I suspect the problem is perhaps with symlinks.  The file in question has these properties in the container immediately after `npm install`:
```
lrwxrwxrwx 1 node node 21 Aug  2 17:28 tsc -> ../typescript/bin/tsc
```
In Windows, the file is a 0 byte file.  If you open the folder locally in VS Code, it doesn't see anything in `node_modules/.bin`.

While disabling `Execute in WSL` does fix the issue, it means one cannot reasonably use that feature since many dev projects create symlinks.

**Comment by MicahZoltu** [comment]
@chrmarti Do you have a link to the PR where this was fixed?

**Comment by chrmarti** [comment]
Fix is available in Dev Containers 0.308.0-pre-release.

@MicahZoltu The source of the Dev Containers extension is not public. There are some notes on licensing if you are curious: https://code.visualstudio.com/docs/remote/faq#_license-and-privacy.

The Dev Containers CLI used by the extension on the other hand is OSS: https://github.com/devcontainers/cli. This fix happened to be in the extension.
